hatch_pip_compile/plugin.py

```diff
@@ -173,24 +173,19 @@ def _pip_sync_cli(self) -> None:
         self.virtual_env.platform.check_command(cmd)
         if not self.dependencies:
             self._piptools_lock_file.unlink()
+        if not self.skip_install:
+            if self.dev_mode:
+                super().install_project_dev_mode()
+            else:
+                super().install_project()
 
-    def install_project(self):
```

Comment:
I'm *pretty* sure that not overriding `install_project` anymore causes some regression for the case of pip-sync. We had some pretty complex workarounds before and they're mostly just gone now :D

https://github.com/pypa/hatch/blob/52a81d80cd03bfb90d99a5837c778971ea0c8c22/src/hatch/cli/application.py#L117
https://github.com/pypa/hatch/blob/52a81d80cd03bfb90d99a5837c778971ea0c8c22/src/hatch/cli/application.py#L131

Per this logic I expect that the project will be installed and then deleted during sync **only for the initial installation**. Then if the hash changes and deps are **re-**installed, it might work ok.

Reply:
Hm I couldn't reproduce any error case, but I still suspect it somehow :/
Too tired to think atm

Reply:
I couldn't produce any errors either - but decided to revert the change to be safe. There doesn't seem to be any perceivable performance penalty